docs/self-managed/setup/deploy/azure/azure-aks/terraform-setup.md

```diff
@@ -0,0 +1,465 @@
+
+```bash reference
+https://github.com/camunda/camunda-deployment-references/blob/main/azure/common/procedure/storage-account/storage-account-tf-init.sh
+```
+
+Terraform will connect to the Azure storage container to manage the state file, ensuring remote and persistent storage.
+
+### AKS cluster module setup
+
+This module establishes the foundational configuration for Azure access and Terraform.
+
+We will utilize [Terraform modules](https://developer.hashicorp.com/terraform/language/modules), which allow us to abstract resources into reusable components, streamlining our infrastructure management and following Terraform best practices.
+
+The reference architecture comes with an example module implementation of the [AKS cluster](https://github.com/camunda/camunda-deployment-references/blob/main/azure/modules/aks-cluster/) and offers a robust starting point for deploying an AKS cluster. It is highly recommended to review this module prior to implementation to understand its structure and capabilities.
+
+The module will be locally sourced, meaning within your cloned repository you can make adjustments to the AKS module directly, and the changes will immediately apply to your setup.
+
+#### Set up the AKS cluster module
+
+1. The `cluster.tf` in your chosen reference contains a basic setup referencing a local Terraform module with the cluster configuration. The following shows said file, which you can adjust within your cloned setup to suit your needs.
+
+```hcl reference
+https://github.com/camunda/camunda-deployment-references/blob/main/azure/modules/aks/cluster.tf
+```
+
+2. Configure user access to the cluster. By default, the identity that creates the Azure AKS cluster has administrative access via role-based access control (RBAC).
+
+<details>
+  <summary>Grant cluster access to other users</summary>
+  <p>
+
+To grant other users access to the cluster, you can assign them Azure roles using Azure Active Directory (Entra ID) integration and Kubernetes RBAC.
+
+This architecture does not currently enable Entra ID integration, but you can manage Kubernetes access through the `az role assignment` and `kubectl` tools:
+
+1. Ensure the user has access to the Azure resource group or subscription, typically via the `Azure Kubernetes Service RBAC Cluster Admin` or `Contributor` role.
+
+2. Then, apply Kubernetes RoleBindings or ClusterRoleBindings to control in-cluster permissions:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: reader-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: view
+subjects:
+  - kind: User
+    name: "<user-object-id>"
+    apiGroup: rbac.authorization.k8s.io
+```
+
+Replace `<user-object-id>` with the Azure AD object ID of the user.
+
+For more detailed instructions, refer to the [AKS access control documentation](https://learn.microsoft.com/azure/aks/manage-azure-rbac).
+
+  </p>
+</details>
```

Comment:
isn't this more a post execution item?

At this point it's not actionable for the user and they can't do anything with it.